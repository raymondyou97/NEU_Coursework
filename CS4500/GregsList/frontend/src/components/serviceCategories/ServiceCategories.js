import React from 'react';

import '../../styles/serviceCategories.css';

class ServiceCategories extends React.Component {
  componentDidMount() {
    this.props.fetchServiceCategories();
  }

  updateTitle = e => {
    this.props.updateTitle(e.target.value);
  };

  updateDescription = e => {
    this.props.updateDescription(e.target.value);
  };

  deleteServiceCategory = id => {
    this.props.deleteServiceCategory(id);
  };

  editServiceCategory = (id, title, description) => {
    this.props.updateId(id);
    this.props.updateTitle(title);
    this.props.updateDescription(description);
  };

  render() {
    const { serviceCategory } = this.props;

    return (
      <div>
        <h3 className="service-category-title">Service Categories</h3>
        <table className="service-category-table">
          <tbody className="service-category-table-body">
            <tr className="service-category-columns">
              <td>Title</td>
              <td>Description</td>
              <td></td>
            </tr>
            <tr className="service-category-inputs">
              <td>
                <input
                  className="service-category-title-input"
                  type="text"
                  placeholder="Title"
                  value={serviceCategory.title}
                  onChange={this.updateTitle}
                />
              </td>
              <td>
                <input
                  className="service-category-description-input"
                  type="text"
                  placeholder="Description"
                  value={serviceCategory.description}
                  onChange={this.updateDescription}
                />
              </td>
              <td className="service-category-input-buttons">
                <button
                  className="service-category-add-button"
                  onClick={this.props.createServiceCategory}
                >
                  Add
                </button>
                <button
                  className="service-category-update-button"
                  onClick={this.props.updateServiceCategory}
                >
                  Update
                </button>
              </td>
            </tr>
            {this.props.serviceCategories.map(serviceCategory => {
              const { id, title, description } = serviceCategory;
              return (
                <tr className="service-category-details" key={id}>
                  <td onClick={() => this.props.showDetails(id)}>{title}</td>
                  <td onClick={() => this.props.showDetails(id)}>
                    {description}
                  </td>
                  <td className="service-category-buttons">
                    <button
                      className="service-category-delete-button"
                      onClick={() => this.deleteServiceCategory(id)}
                    >
                      Delete
                    </button>
                    <button
                      className="service-category-edit-button"
                      onClick={() =>
                        this.editServiceCategory(id, title, description)
                      }
                    >
                      Edit
                    </button>
                  </td>
                </tr>
              );
            })}
          </tbody>
        </table>
      </div>
    );
  }
}

export default ServiceCategories;
