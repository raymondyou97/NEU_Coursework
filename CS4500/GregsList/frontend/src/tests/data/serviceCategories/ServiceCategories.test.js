'use es6';

import React from 'react';
import ServiceCategories from '../../../components/serviceCategories/ServiceCategories';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { SERVICE_CATEGORY_LIST } from './ServiceCategoryData';

configure({ adapter: new Adapter() });

test('Snapshot renders', () => {
  const serviceCategory = {
    title: '',
    description: '',
  };

  const component = renderer.create(
    <ServiceCategories
      fetchServiceCategories={() => {}}
      serviceCategory={serviceCategory}
      updateId={() => {}}
      serviceCategories={SERVICE_CATEGORY_LIST}
      updateTitle={() => {}}
      updateDescription={() => {}}
      createServiceCategory={() => {}}
      showDetails={() => {}}
      deleteServiceCategory={() => {}}
      updateServiceCategory={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Users component updates when there is user data', () => {
  const component = renderer.create(
    <ServiceCategories
      fetchServiceCategories={() => {}}
      serviceCategory={{
        title: 'Car Services',
        description: 'All things related to taking care of your car',
      }}
      updateId={() => {}}
      serviceCategories={SERVICE_CATEGORY_LIST}
      updateTitle={() => {}}
      updateDescription={() => {}}
      createServiceCategory={() => {}}
      showDetails={() => {}}
      deleteServiceCategory={() => {}}
      updateServiceCategory={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});

test('Initial DOM contains correct number of rows', () => {
  const serviceCategory = {
    title: '',
    description: '',
  };

  const component = shallow(
    <ServiceCategories
      fetchServiceCategories={() => {}}
      serviceCategory={serviceCategory}
      updateId={() => {}}
      serviceCategories={SERVICE_CATEGORY_LIST}
      updateTitle={() => {}}
      updateDescription={() => {}}
      createServiceCategory={() => {}}
      showDetails={() => {}}
      deleteServiceCategory={() => {}}
      updateServiceCategory={() => {}}
    />
  );

  expect(component.find('tr').length).toEqual(4);
});

test('Users component input value changes if selected user specified', () => {
  const serviceCategory = {
    title: 'Car Services',
    description: 'All things related to taking care of your car',
  };

  const component = shallow(
    <ServiceCategories
      fetchServiceCategories={() => {}}
      serviceCategory={serviceCategory}
      updateId={() => {}}
      serviceCategories={SERVICE_CATEGORY_LIST}
      updateTitle={() => {}}
      updateDescription={() => {}}
      createServiceCategory={() => {}}
      showDetails={() => {}}
      deleteServiceCategory={() => {}}
      updateServiceCategory={() => {}}
    />
  );

  expect(component.find('.service-category-title-input').props().value).toEqual(
    'Car Services'
  );
  expect(
    component.find('.service-category-description-input').props().value
  ).toEqual('All things related to taking care of your car');
});
