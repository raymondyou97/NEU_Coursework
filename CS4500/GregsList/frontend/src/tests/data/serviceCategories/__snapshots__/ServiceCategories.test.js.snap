// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<div>
  <h3
    className="service-category-title"
  >
    Service Categories
  </h3>
  <table
    className="service-category-table"
  >
    <tbody
      className="service-category-table-body"
    >
      <tr
        className="service-category-columns"
      >
        <td>
          Title
        </td>
        <td>
          Description
        </td>
        <td />
      </tr>
      <tr
        className="service-category-inputs"
      >
        <td>
          <input
            className="service-category-title-input"
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value=""
          />
        </td>
        <td>
          <input
            className="service-category-description-input"
            onChange={[Function]}
            placeholder="Description"
            type="text"
            value=""
          />
        </td>
        <td
          className="service-category-input-buttons"
        >
          <button
            className="service-category-add-button"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="service-category-update-button"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-category-details"
      >
        <td
          onClick={[Function]}
        >
          Car Services
        </td>
        <td
          onClick={[Function]}
        >
          All things related to taking care of your car
        </td>
        <td
          className="service-category-buttons"
        >
          <button
            className="service-category-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-category-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-details"
      >
        <td
          onClick={[Function]}
        >
          Home Care
        </td>
        <td
          onClick={[Function]}
        >
          For all things related to taking care of the home
        </td>
        <td
          className="service-category-buttons"
        >
          <button
            className="service-category-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-category-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Users component updates when there is user data 1`] = `
<div>
  <h3
    className="service-category-title"
  >
    Service Categories
  </h3>
  <table
    className="service-category-table"
  >
    <tbody
      className="service-category-table-body"
    >
      <tr
        className="service-category-columns"
      >
        <td>
          Title
        </td>
        <td>
          Description
        </td>
        <td />
      </tr>
      <tr
        className="service-category-inputs"
      >
        <td>
          <input
            className="service-category-title-input"
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value="Car Services"
          />
        </td>
        <td>
          <input
            className="service-category-description-input"
            onChange={[Function]}
            placeholder="Description"
            type="text"
            value="All things related to taking care of your car"
          />
        </td>
        <td
          className="service-category-input-buttons"
        >
          <button
            className="service-category-add-button"
            onClick={[Function]}
          >
            Add
          </button>
          <button
            className="service-category-update-button"
            onClick={[Function]}
          >
            Update
          </button>
        </td>
      </tr>
      <tr
        className="service-category-details"
      >
        <td
          onClick={[Function]}
        >
          Car Services
        </td>
        <td
          onClick={[Function]}
        >
          All things related to taking care of your car
        </td>
        <td
          className="service-category-buttons"
        >
          <button
            className="service-category-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-category-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
      <tr
        className="service-category-details"
      >
        <td
          onClick={[Function]}
        >
          Home Care
        </td>
        <td
          onClick={[Function]}
        >
          For all things related to taking care of the home
        </td>
        <td
          className="service-category-buttons"
        >
          <button
            className="service-category-delete-button"
            onClick={[Function]}
          >
            Delete
          </button>
          <button
            className="service-category-edit-button"
            onClick={[Function]}
          >
            Edit
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
